### PR TITLE
Send all button press events for Sinope SW2500ZB/DM2500ZB

### PR DESCRIPTION
**Pull request by bassdr**
## Proposed change

Added support for both press and release button events in order to properly support short and long press events.
Also added the pressed button and text description in the event arguments to simplify writing automations processing it.

ZHA now send both when we start pressing the Switch's up or down button, and when we release. Before, we were only sending events when a button was released after being pressed for a long time, or if double pressed, limiting possibilities. Now all events are sent: pressed, short released, long released and double pressed.

Note that I also added two extra arguments the the send `zha_event` because it was a bit hard to extract which button was pressed (up or down). It's now a new attribute `{button: turn_on}` or `{button: turn_off}` and a `description` that contains the `ActionButton` name (Pressed_on, Released_off, Double_on, etc.)

## Additional information

I have been using this as a custom quirks for a while. I have two dimmer switches DM2500ZB dimmer switches and I'm emulating a 3-way dimmers with the downstairs dimmer using an HA automation to control upstairs dimmer that has the stairs lights connect to it. It works fine.

I also have two SW2500ZB, the events are also sent as expected.

## Checklist

- [x] The changes are tested and work correctly
- [x] `pre-commit` checks pass / the code has been formatted using Black
- [x] Tests have been added to verify that the new code works

### CODE REVIEW

**Comment by bassdr** [comment]
@claudegel FYI.

File: zhaquirks/sinope/__init__.py
Comment (TheJulianJES) [inline zhaquirks/sinope/__init__.py]: This would break existing automations with the "long press" trigger, right?

Comment (dmulcahey) [reply zhaquirks/sinope/__init__.py]: Yes

Comment (bassdr) [reply zhaquirks/sinope/__init__.py]: Indeed. The automation using them will show an error. Easy fix tho.

Is there a deprecation mechanism in place in ha? Or worse case I guess we can alias? It's really when we release the button that it sends either a long or short press event :/

Comment (TheJulianJES) [reply zhaquirks/sinope/__init__.py]: There is not, unfortunately. What do you think about changing this back for now? Only real difference between them is another UI text for the end-user.

Comment (bassdr) [reply zhaquirks/sinope/__init__.py]: If it's the only thing that breaks backward compatibility, I'm fine. If someone was  directly listening too zha_event, there are more breaking changes tho... Since the command changed too. Not sure if it's acceptable?

Comment (bassdr) [reply zhaquirks/sinope/__init__.py]: I think my last commit fixes this concern. I'll leave you the final word.

**Comment by bassdr** [comment]
I've tested my latest changes in HA and it works fine.

File: zhaquirks/sinope/sensor.py
Comment (TheJulianJES) [inline zhaquirks/sinope/sensor.py]: Please revert to using absolute imports everywhere. It's easier to use them as custom quirks then.
Background: https://github.com/zigpy/zha-device-handlers/pull/827

Comment (bassdr) [reply zhaquirks/sinope/sensor.py]: It's quite the opposite. Without them I can't use this as a quirk. Since there are modifications to the `__init__`. Without them you'll be using the wrong init file. I definitely challenge the call to use absolute imports for current module. I don't understand what's easier either. My setup to include this as a quirks is very simple.

Comment (TheJulianJES) [reply zhaquirks/sinope/sensor.py]: Ah, so you're importing all these via the "custom quirks" feature.
If you do bigger modifications (involving multiple classes/files), it's ideal to directly modify the `zhaquirks` package.

We need absolute imports, as we want users to be able to easily modify a single quirk file (that imports from other files), without needing to load everything as custom quirks. For example, to easily add a single model to `MODELS_INFO`.
So, these changes need to be reverted before we can merge it.

Comment (bassdr) [reply zhaquirks/sinope/sensor.py]: done

**Review by TheJulianJES — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**
